sdk/src/main/java/com/hedera/hashgraph/sdk/Network.java

```diff
@@ -159,167 +130,94 @@ NodeAddress getAddressBook(AccountId nodeAccountId) {
+        var nodes = getNumberOfMostHealthyNodes(getNumberOfNodesForRequest());
+        var nodeAccountIds = new ArrayList<AccountId>(nodes.size());
 
-        List<AccountId> resultNodeAccountIds = new ArrayList<>();
-
-        for (int i = 0; i < getNumberOfNodesForTransaction(); i++) {
-            resultNodeAccountIds.add(nodes.get(i).accountId);
+        for (var node : nodes) {
+            nodeAccountIds.add(node.getAccountId());
         }
 
-        lock.release();
-
-        return resultNodeAccountIds;
-    }
-
-    void setMaxNodesPerTransaction(int maxNodesPerTransaction) {
-        this.maxNodesPerTransaction = maxNodesPerTransaction;
+        return nodeAccountIds;
     }
 
-    int getMaxNodeAttempts() {
-        return maxNodeAttempts;
-    }
-
-    void setMaxNodeAttempts(int maxNodeAttempts) {
-        this.maxNodeAttempts = maxNodeAttempts;
-    }
-
-    Duration getNodeWaitTime() {
-        return nodeWaitTime;
-    }
-
-    void setNodeWaitTime(Duration nodeWaitTime) {
-        this.nodeWaitTime = nodeWaitTime;
-        for (var node : nodes) {
-            node.setWaitTime(nodeWaitTime.toMillis());
-        }
+    public Network setMaxNodesPerRequest(int maxNodesPerRequest) {
+        this.maxNodesPerRequest = maxNodesPerRequest;
+        return this;
     }
 
-    int getNumberOfNodesForTransaction() {
-        if (maxNodesPerTransaction != null) {
-            return Math.min(maxNodesPerTransaction, nodes.size());
+    public int getNumberOfNodesForRequest() {
+        if (maxNodesPerRequest != null) {
+            return Math.min(maxNodesPerRequest, nodes.size());
         } else {
             return (nodes.size() + 3 - 1) / 3;
         }
     }
 
-    void close(Duration timeout) throws TimeoutException {
-        try {
-            lock.acquire();
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        }
-
-        var stopAt = Instant.now().getEpochSecond() + timeout.getSeconds();
-
-        for (var node : nodes) {
-            if (node.channel != null) {
-                node.channel = node.channel.shutdown();
-            }
-        }
-
-        for (var node : nodes) {
-            if (stopAt - Instant.now().getEpochSecond() == 0) {
-                lock.release();
-                throw new TimeoutException("Failed to properly shutdown all channels");
-            }
-
-            if (node.channel != null) {
-                try {
-                    node.channel.awaitTermination(stopAt - Instant.now().getEpochSecond(), TimeUnit.SECONDS);
-                } catch (InterruptedException e) {
-                    lock.release();
-                    throw new RuntimeException(e);
-                }
-            }
-        }
-
-        nodes.clear();
-        networkNodes.clear();
-        network.clear();
-
-        lock.release();
+    public Node getNodeForAccountId(AccountId nodeAccountId) {
```

Comment:
Methods in `Network` don't need to be public. By exposing them it makes them part of the permanent API that you can't change.

Reply:
Maybe I'm misunderstanding, but since `Network` is package-private then all the methods are at most package-private as well even when they're declared `public`. The big reason I added an explicit scope to most methods is because I wanted to make it a conscious effort to determine if a method should be accessible by other classes or not. It is far too easy to mistakenly add a new method as package private even when it should be private and/or change a package-private method's behavior without realizing other classes rely on that method's functionality.

Reply:
In general, you should always use the smallest scope possible until you have a need to widen it. Since the SDK has every class in one package (not ideal, IMO), everything should be private or package private unless the client directly uses it.

When you have a package private class with public methods, if another class that's public inherits from that class suddenly those methods are now public for use by everyone. The developer who extended the class may not have intended that since he didn't closely examine the other code that he's not touching. Here, you don't currently have another class inherit from `Network` so it's not a problem at this time, but that could change in the future and they may get mistakenly exposed. Better to reduce the scope until needed.